classes/Transformer.php

```diff
@@ -0,0 +1,102 @@
+<?php
+class Transformer
+{
+	private $config;
+	private $splitter;
+
+	/**
+	 * Transformer constructor.
+	 * @param int $id
+	 */
+	public function __construct(int $id = 1) {
+		$service = new Service($id);
+		$this->config = $service->config;
+
+		$splitter = $service->config['splitter'] ?? null;
+		$this->splitter = !empty($splitter) ? $splitter . ' ' : '';
+	}
+
+	/**
+	 * @param int $input
+	 * @return string
+	 */
+	public function fullTransform(int $input): string {
+		$result = array_filter([
+			$this->divTransform($input, false),
+			$this->attachTransform($input, false)
+		]);
+		$mergedResult = $this->resultMerge($result, $input, true);
+		return $mergedResult . $this->sumTransform($input);
+	}
+
+	/**
+	 * @param int $input
+	 * @param bool $needNumber
+	 * @return string
+	 */
+	public function divTransform(int $input, bool $needNumber = true): string {
```

Comment:
You need to rewrite here and later function to avoid argument `$needNumber`. This specific case should be handled on higher level of abstraction.